src/main/java/spoon/reflect/meta/impl/AbstractRoleHandler.java

```diff
@@ -134,8 +139,64 @@ public ContainerKind getContainerKind() {
 			return asList(element);
 		};
 
-		public <W, X> java.util.List<X> asList(W element) {
-			return Collections.<X>singletonList(getValue(element));
+		public <W, X> java.util.List<X> asList(W e) {
```

Comment:
I think there's already a lot of mixed responsibilities in this file and adding a new anonymous class here increases the complexity. So I'm ok with your changes but I'm in favor of extracting the static classes from this file to put them in their own file: I think it will help the future maintaining. WDYT?